src/components/sbb-timetable-occupancy-icon/sbb-timetable-occupancy-icon.ts

```diff
@@ -0,0 +1,89 @@
+import { CSSResult, html, LitElement, PropertyValues, TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import style from './sbb-timetable-occupancy-icon.scss?lit&inline';
+import { OccupancyEnum } from '../core/timetable';
+import { documentLanguage, HandlerRepository, languageChangeHandlerAspect } from '../core/eventing';
+import { i18nOccupancy } from '../core/i18n';
+import '../sbb-icon';
+
+@customElement('sbb-timetable-occupancy-icon')
+export class SbbTimetableOccupancyIcon extends LitElement {
+  public static override styles: CSSResult = style;
+
+  /** Wagon occupancy. */
+  @property() public occupancy!: OccupancyEnum;
+
+  /** Negative coloring variant flag. */
+  @property({ type: Boolean }) public negative: boolean = false;
+
+  /** The icon name which will be rendered. */
+  @state() private _iconName: string;
+
+  @state() private _currentLanguage = documentLanguage();
+
+  private _handlerRepository = new HandlerRepository(
+    this,
+    languageChangeHandlerAspect((l) => (this._currentLanguage = l)),
+  );
+
+  private _setIconName(): void {
+    if (!this.occupancy) {
+      return;
+    }
+
+    if (window.matchMedia('(forced-colors: active)').matches) {
+      // high contrast
+      this._iconName = `utilization-${this._transformOccupancy()}-high-contrast`;
+    } else if (
+      this.hasAttribute('negative') ||
+      window.matchMedia('(prefer-color-scheme: dark)').matches
+    ) {
+      // dark
+      this._iconName = `utilization-${this._transformOccupancy()}-negative`;
+    } else {
+      this._iconName = `utilization-${this._transformOccupancy()}`;
+    }
+  }
+
+  private _transformOccupancy(): string {
+    return !this.occupancy || this.occupancy.toLowerCase() === 'unknown'
+      ? 'none'
+      : this.occupancy.toLowerCase();
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    this._setIconName();
+    window.matchMedia('(forced-colors: active)').onchange = () => this._setIconName();
```

Comment:
what happens to the eventlisteners if the component gets reconnected or gets disconnected? This looks like a memory issue. The solution is maybe easy to just unsubscribe in disconnectedcallback?

Reply:
Cleared in disconnected callback, yes, i think it should be enough